buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -475,10 +474,11 @@ public void goToNextVersion() {
             throw new TestClustersException("Ran out of versions to go to for " + this);
         }
         logToProcessStdout("Switch version from " + getVersion() + " to " + distributions.get(currentDistro + 1).getVersion());
-        stop(false);
         currentDistro += 1;
-        setting("node.attr.upgraded", "true");
-        start();
+        setting("node.attr.upgraded", "true", PropertyNormalization.IGNORE_VALUE);
+        if (esProcess != null) {
```

Comment:
Is the null check actually needed ?  
I don't think we have a use-case for going to the next version without starting first ?

Reply:
This happens when the the tests are cached. When they are, we don't start the clusters but we still run the "upgrade" tasks.

Reply:
So we still need to perform the "upgrade" in the sense that we need to bump the node's `currentDistro` since that is being tracked as a input for the subsequent test tasks.